modules/backend/traits/ModelOptions.php

```diff
@@ -0,0 +1,97 @@
+<?php namespace Backend\Traits;
+
+use Lang;
+use ApplicationException;
+
+/**
+ * Model Options Trait
+ * Implements helper methods useful to extract the options from a given field
+ * that is used in the List and Form widgets.
+ */
+trait ModelOptions
+{
+    /**
+     * Looks at the model for defined options.
+     *
+     * @param $field
```

Comment:
There's still a conflict here in that Form is passing a `Backend\Classes\FormField` instance and Lists is passing a `October\Rain\Database\Model` instance - these two options are not identical to each other so they need to be treated differently in the method itself.

Reply:
Yeah, I knew this but it worked fine so didn't want to change it. I'll see what can be done...

Reply:
It works fine for now but there are code branches that treat $field as a Field instead of a model which can cause issues if those branches are reached under the wrong conditions.

Reply:
@LukeTowers I do not think of a better way to make getOptionsFromModel method more generic as Lists and Forms use different classes for their fields i.e. Lists uses ListColumn and Form uses FormField. One solution would be to include the following code:
```
if ($this instanceof \Backend\Widgets\Lists) {
    $field = $this->getColumn($fieldName);
} else if ($this instanceof \Backend\Widgets\Form) {
    $field = $this->getField($fieldName);
}
```
at the top of the method to generalise the field variable. The other alternative was allowing mixed variables to be passed to the method and I've done exactly this in my last commit: the form sends a variable of type FormField and the list sends variable of type ListColumn. I've modified the internal code in the getOptionsFromModel so that it can handle both these field types ideally: the issue I faced was that ListColumn did not have a $value variable so I modified the code to use the method getValueFromData which is common to both types and has similar method signature. So I've generalised this as much as possible. Let me know if this is alright, or suggest me a better way of solving this.